sync/syncer.go

```diff
@@ -256,9 +257,10 @@ func (s *Syncer) GetFullBlocks(blockIds []types.BlockID) []*types.Block {
 }
 
 func (s *Syncer) BlockSyntacticValidation(block *types.Block) ([]*types.AddressableSignedTransaction, []*types.ActivationTx, error) {
-	if err := s.confirmBlockValidity(block); err != nil {
-		s.Error("block %v validity failed %v", block.ID(), err)
-		return nil, nil, errors.New(fmt.Sprintf("failed derefrencing block data %v %v", block.ID(), err))
+
+	//block eligibility
+	if eligable, err := s.BlockSignedAndEligible(block); err != nil || !eligable {
+		return nil, nil, errors.New(fmt.Sprintf("block %v eligablety check failed %v", block.ID(), err))
 	}
 
 	//data availability
```

Comment:
You said that you has a way to also put data availability under BlockSignedAndEligible(block), so that the entire BlockSyntacticValidation() process would be a black box to syncer

Reply:
i talked with yuval about refactoring the sync code, i think we add lots of logic in a short while a step back would be good
just need to prioritise over bugs.